README.md

```diff
@@ -1,69 +1,97 @@
 # multiplex
+Translates a JSON with multi-value parameters into a JSON with single-value parameters.
 
 ## Introduction
 When running a benchmark, it is often desirable to run it multiple ways, changing how the options are used, so you can get a full characterization of whatever you are testing.  For example, one might use several message sizes for a network test, or use different IO-engines for a storage test.  However, when executing a benchmark, most of them expect to have exactly one value for each option, like a message-size of just "16k" or an IO-engine of just "libaio".  Multiplex will translate your multi-value options into a list of multiple command-line statements that your benchmark understands.
 
+## Usage
+```
+./multiplex.py --input JSON/mv-params-input.json > /path/to/bench-params.json
+```
+
 ## Input and Output
-Multiplex requires a JSON from STDIN with the following format:
-<pre>
+Multiplex requires a JSON file with the following format:
+```
 {
     "common": [
-        { "arg": "ioengine", "values": [ "sync", "aio" ] },
-        { "arg": "runtime", "values": [ "60s", "120s" ] }
+    {
+        "name": "global",
+        "params": [
+        { "arg": "bs", "vals": [ "4k", "8k" ] },
+        { "arg": "rw", "vals": [ "read", "write" ] }
+        ]
+    }
     ],
     "sets": [
-        [
-            { "arg": "bs", "values": [ "4k", "8k" ] },
-            { "arg": "rw", "values": [ "read", "write" ] }
-        ],
-        [
-            { "arg": "bs", "values": [ "16k", "32k" ] },
-            { "arg": "rw", "values": [ "randread", "randwrite" ] },
-            { "arg": "runtime", "values": [ "30s" ] }
-        ]
+    [
+        { "common": "global" },
+        { "arg": "ioengine", "vals": [ "sync" ] }
+    ]
     ]
 }
-</pre>
-
-The common section is optonal, and the sets section is required.  At least one set is required.
-A set of "multi-value parameters" are included in each set.  Each set, combined with the multi-value paramters in "common" (if it exists), will be used to expand to a new set of single-value parameters to STDOUT, like:
+```
 
-<pre>
+The `common` section is optional, and the `sets` section is required.  At least one set is required.
```

Comment:
We need to look at the schema...at the moment I think that the common section is required, but it can be empty.  This would probably be a good starting point for introducing you to the schema stuff.  Also, I think the schema probably doesn't match up to what we have decided to do so it needs updating anyway.